### DISCUSSION TITLE
Question: Why do code examples not follow lower_case variable names and UPPER_CASE constants?

### DISCUSSION THREAD

**Discussion by couling** [prometheus/client_python#869] [Q&A]
Apologies for asking about a really boring topic.

I noticed in the README, metrics stored globally with upper case naming:
https://github.com/prometheus/client_python/blob/d6e08e3ad90f0f65f95c59f0776052046e6e26e0/README.md?plain=1#L19

This is a bit confusing for many readers because across multiple languages ([including python](https://peps.python.org/pep-0008/#constants)) `UPPER_CASE` names are use to denote a constant (a value that does not change) and `lower_case` names are used for variables (that do change). 

Likewise in the code base there are things like `REGISTRY` that are uppercase for non-constant items: https://github.com/prometheus/client_python/blob/d6e08e3ad90f0f65f95c59f0776052046e6e26e0/prometheus_client/registry.py#L168 

Is there a specific reason that these examples a reusing upper case names even though it conflicts with common standard and doesn't match the metric name?

I'm just asking because if there is a good reason, this might actually be worth documenting.

Apologies again for the trollish sounding question.

**Comment by csmarchbanks** [comment]
These decisions were made before I worked on this project, but I believe it is because the metric object should not change, e.g. you should not replace `REQUEST_TIME` with a new `Summary`. Similarly the default registry should not be replaced with another registry. You can interact with it, but should not overwrite it.

There could be a different reason, but that is my thought.

**Reply by couling** [reply]
Hmmm. It may have been the reason, but it's a misunderstanding of "constant".  For example a list object that is never replaced, but continually modified appending and clearing is not constant.
